### PR TITLE
CAM-11416: doc(feel/spin): add feel engine spin integration usage

### PR DESCRIPTION
**Pull request by koevskinikola**
[CAM-11416](https://app.camunda.com/jira/browse/CAM-11416)

* Document the Spin integration for the Scala FEEL Engine. Add sections for location, usage and features.

Related to CAM-11416

Note: I also added a note in the FEEL reference page that points to the Legacy FEEL engine pages. I think it makes more sense since we already point to the PEC propert there.

### CODE REVIEW

File: content/user-guide/dmn-engine/feel/spin-integration.md [hunk 1]
```diff
@@ -8,13 +8,124 @@ menu:
...
 ---
 
-* How it works with the process-engine-plugin
-    * have copyable maven coordinates
-* Explain how it works
-    * needs specific Spin types
-    * show some code (setting a json variable)
-* To what FEEL types are json/xml variables mapped? => link to external docu
\ No newline at end of file
+## Where can the Spin integration be found
```
Comment (tasso94) [inline content/user-guide/dmn-engine/feel/spin-integration.md]: image
Cann we add an introduction paragraph between "FEEL Engine Spin Integration" and "Where can the Spin integration be found"? I think its better style to not have two headlines directly follow each other.

File: content/user-guide/dmn-engine/feel/type-handling.md [hunk 2]
```diff
@@ -14,4 +14,13 @@ menu:
 
 * Add table for supported input types
     * this can also include spin types => link to the spin integration page
-* Add table for supported output types
\ No newline at end of file
+
+### Spin data types
```
Comment (tasso94) [inline content/user-guide/dmn-engine/feel/type-handling.md]: Let's remove the whole section since it is covered in PR #456

Comment (koevskinikola) [reply content/user-guide/dmn-engine/feel/type-handling.md]: :+1:

File: content/user-guide/dmn-engine/feel/spin-integration.md (hunk 1 above)
Comment (koevskinikola) [reply content/user-guide/dmn-engine/feel/spin-integration.md]: :+1:

**Pull request merged**